extensions/big-data-cluster/src/bigDataCluster/dialog/addControllerDialog.ts

```diff
@@ -7,40 +7,74 @@
 
 import * as azdata from 'azdata';
 import * as nls from 'vscode-nls';
-import { getEndPoints, ControllerError } from '../controller/clusterControllerApi';
+import { ClusterController, ControllerError } from '../controller/clusterControllerApi';
 import { ControllerTreeDataProvider } from '../tree/controllerTreeDataProvider';
 import { TreeNode } from '../tree/treeNode';
 import { showErrorMessage } from '../utils';
+import { AuthType } from '../constants';
 
 const localize = nls.loadMessageBundle();
 
+const basicAuthDisplay = localize('basicAuthName', "Basic");
+const integratedAuthDisplay = localize('integratedAuthName', "Windows Authentication");
+
+function getAuthCategory(name: AuthType): azdata.CategoryValue {
+	if (name === 'basic') {
+		return { name: name, displayName: basicAuthDisplay };
+	}
+	return { name: name, displayName: integratedAuthDisplay };
+}
+
 export class AddControllerDialogModel {
 
 	private _canceled = false;
-
+	private _authTypes: azdata.CategoryValue[];
 	constructor(
 		public treeDataProvider: ControllerTreeDataProvider,
 		public node?: TreeNode,
 		public prefilledUrl?: string,
+		public prefilledAuth?: azdata.CategoryValue,
 		public prefilledUsername?: string,
 		public prefilledPassword?: string,
 		public prefilledRememberPassword?: boolean
 	) {
 		this.prefilledUrl = prefilledUrl || (node && node['url']);
+		this.prefilledAuth = prefilledAuth;
+		if (!prefilledAuth) {
+			let auth = (node && node['auth']) || 'basic';
+			this.prefilledAuth = getAuthCategory(auth);
+		}
 		this.prefilledUsername = prefilledUsername || (node && node['username']);
 		this.prefilledPassword = prefilledPassword || (node && node['password']);
 		this.prefilledRememberPassword = prefilledRememberPassword || (node && node['rememberPassword']);
 	}
 
-	public async onComplete(url: string, username: string, password: string, rememberPassword: boolean): Promise<void> {
+	public get authCategories(): azdata.CategoryValue[] {
+		if (!this._authTypes) {
+			this._authTypes = [getAuthCategory('basic'), getAuthCategory('integrated')];
+		}
+		return this._authTypes;
+	}
+
+	public async onComplete(url: string, auth: AuthType, username: string, password: string, rememberPassword: boolean): Promise<void> {
 		try {
+
+			if (auth === 'basic') {
+				// Verify username and password as we can't make them required in the UI
+				if (!username) {
+					throw new Error(localize('err.controller.username.required', "Username is required"));
+				} else if (!password) {
+					throw new Error(localize('err.controller.password.required', "Password is required"));
+				}
```

Comment:
What about just disabling the OK button until those fields are filled? Not sure if we can hook into the text updated events for the text box components though currently.

Also this means we won't have the red * anymore right? That's unfortunate if so, did you look into keeping that functionality?

Reply:
It is unfortunate. As mentioned I'm not aware of a way to do this without bigger changes. For now, I'm going to go with current implementation as I think it's fairly understandable.